src/main/java/io/cosmosoftware/kite/steps/ScreenshotStep.java

```diff
@@ -0,0 +1,28 @@
+package io.cosmosoftware.kite.steps;
+
+import io.cosmosoftware.kite.exception.KiteTestException;
+import io.cosmosoftware.kite.report.Reporter;
+import org.openqa.selenium.WebDriver;
+
+import static io.cosmosoftware.kite.util.ReportUtils.saveScreenshotPNG;
+import static io.cosmosoftware.kite.util.ReportUtils.timestamp;
+
+public class ScreenshotStep extends TestStep {
+
+  public ScreenshotStep(StepParams params) {
+    super(params);
+    setStepPhase(StepPhase.ALL);
+  }
+
+  @Override
+  protected void step() throws KiteTestException {
+    Reporter.getInstance()
+      .screenshotAttachment(
+        report, "ScreenshotStep_" + timestamp(), saveScreenshotPNG(webDriver));
```

Comment:
There's an exception BrowserCantBeReached or something. THis happens sometimes when taking screenshot, and get thrown as UnHanddled Exception. Might need to check to see whether this is caught in saveScreenshotPNG()

Reply:
that should result in a broken step. I didnt encounter this so far.